iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/TieLineImpl.java

```diff
@@ -124,9 +124,21 @@ public double getB2() {
 
     @Override
     public void remove() {
+        remove(false);
+    }
+
+    @Override
+    public void remove(boolean updateDanglingLines) {
         NetworkImpl network = getNetwork();
         network.getListeners().notifyBeforeRemoval(this);
 
+        if (updateDanglingLines) {
+            danglingLine1.setP0(danglingLine1.getBoundary().getP());
```

Comment:
There should be a change in sign. It should be `setP0(-danglingLine1.getBoundary().getP())` (you can check it looking at `DanglingLineBoundaryImpl.java::getP`). `P0, Q0` is a load (consumption) located at the boundary node, and `boundary.getP` is interpreted as the flow of the line seen at the boundary terminal.